python/ray/_private/client_mode_hook.py

```diff
@@ -1,29 +1,40 @@
 import os
 from contextlib import contextmanager
 from functools import wraps
+import threading
 
 # Attr set on func defs to mark they have been converted to client mode.
 RAY_CLIENT_MODE_ATTR = "__ray_client_mode_key__"
 
 client_mode_enabled = os.environ.get("RAY_CLIENT_MODE", "0") == "1"
 os.environ.update({"RAY_CLIENT_MODE": "0"})
 
-_client_hook_enabled = True
+_client_hook_enabled = threading.local()
+_client_hook_enabled.val = True
+
+def _get_client_hook_enabled():
+    global _client_hook_enabled
+    if not hasattr(_client_hook_enabled, "val"):
+        _client_hook_enabled.val = True
```

Comment:
Just curious. Is this right: when you start a new thread, its copy of `_client_hook_enabled` doesn't have `val` set, hence the need for this function?

Reply:
@DmitriGekhtman Yep!